### PR TITLE
Sparse CSR: Implement unary ufuncs (with 0->0 correspondence)

### PR DESCRIPTION
**Pull request by krshrimali**
This PR attempts to add support for unary ufuncs (with 0->0 correspondence) for Sparse CSR Layout.

Ops supported: `['abs', 'asin', 'asinh', 'atan', 'atanh', 'ceil', 'conj_physical', 'floor', 'log1p', 'neg', 'round', 'sin', 'sinh', 'sign', 'sgn', 'signbit', 'tan', 'tanh', 'trunc', 'expm1', 'sqrt', 'angle', 'isinf', 'isposinf', 'isneginf', 'isnan', 'erf', 'erfinv']`

cc @nikitaved @pearu @cpuhrsch @IvanYashchuk @peterbell10

### CODE REVIEW

**Review by cpuhrsch — CHANGES_REQUESTED** [review]
Thanks Kush! Looking good. 

But, next to the OpInfos you'll also need to update the unary tests within test/test_sparse.py. Please also see the failing checks below.

I'll import in the meanwhile to get signal from internal CI.

**Comment by cpuhrsch** [comment]
sinh for any dtypes and log10 for integer dtypes tests are failing internally.

**Comment by krshrimali** [comment]
> sinh for any dtypes and log10 for integer dtypes tests are failing internally. Will reimport with the last two changes, but this didn't seem addressed by your new commits, so it's likely still relevant.

Thanks, @cpuhrsch for taking a look and updating with the possible failures. I've attempted to fix things here, waiting for the CI to give the green signal. If you could take another look whenever you find the time, that will be great. Thanks a lot for your patience, and help.

**Review by cpuhrsch — APPROVED** [review]
Looks good, let's land it :)

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -4293,8 +4347,6 @@
   variants: function, method
 
 - func: square.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
-  dispatch:
```
Comment (cpuhrsch) [inline aten/src/ATen/native/native_functions.yaml:4296]: Actually, what's the reason you removed this?

**Review by cpuhrsch — CHANGES_REQUESTED** [review]

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (krshrimali) [reply aten/src/ATen/native/native_functions.yaml:4296]: Sorry, should have mentioned before. I'm getting `RuntimeError: derivative for aten::square is not implemented` while adding support for `square`. So, I thought I'll take a look at this in a separate PR if that's alright?

Comment (cpuhrsch) [reply aten/src/ATen/native/native_functions.yaml:4296]: Hm, but this PR doesn't add support square if I read it correctly. Why remove the dispatch for CPU and CUDA here?

Comment (krshrimali) [reply aten/src/ATen/native/native_functions.yaml:4296]: Oh, my bad. Didn't mean to remove it. I'll bring it back.

Comment (krshrimali) [reply aten/src/ATen/native/native_functions.yaml]: Done. Sorry again, @cpuhrsch! :)

**Review by cpuhrsch — APPROVED** [review]
Great job @krshrimali! Thank you!

File: test/test_sparse_csr.py [hunk 2]
```diff
@@ -1122,8 +1122,77 @@ def test_sparse_csr_unary(self, device, dtype, op):
...
+            assert torch.is_tensor(sample.input)
+            # Sparse CSR only supports 2D tensors as inputs
+            # Fail early to prevent silent success with this test
+            if sample.input.ndim != 2:
+                raise ValueError("Expected 2D tensor but got tensor with dimension: {sample.input.ndim}.")
+
+            sample.input = sample.input.to_sparse_csr()
+            expect = op(sample.input, *sample.args, **sample.kwargs)
+
+            out = self.genSparseCSRTensor(sample.input.size(), sample.input._nnz(),
```
Comment (IvanYashchuk) [inline test/test_sparse_csr.py:1148]: In a follow-up, I think we should also test the behavior when `out.shape == (0, 0)`, `out._nnz() == 0` (`out` should be resized correctly in this case), and when `out.shape` is wrong.

**Review by IvanYashchuk — APPROVED** [review]

File: test/test_sparse_csr.py (hunk 2 above)
Comment (krshrimali) [reply test/test_sparse_csr.py:1148]: Acknowledging that I've taken note of this. Thanks, @IvanYashchuk - I'll address this in a follow-up PR :)

**Pull request closed without merging**